codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,286 @@
+     * @return a {@link BulkStringRedisMessage} with {@code null} content.
+     */
+    public static RedisMessage nullBulkString() {
+        return NULL_BULK_STRING;
+    }
+
+    /**
+     * Creates a {@link BulkStringRedisMessage} with empty content.
+     *
+     * @return a {@link BulkStringRedisMessage} with empty content.
+     */
+    public static RedisMessage emptyBulkString() {
+        return EMPTY_BULK_STRING;
+    }
+
+    /**
+     * Creates a {@link BulkStringRedisMessage} for the given {@code content}.
+     * Note that non-empty {@code content} results in {@link io.netty.util.ReferenceCounted} instances.
+     *
+     * @param content the content, must not be {@code null}.
+     * @return a {@link BulkStringRedisMessage} for the given {@code content}
+     */
+    public static RedisMessage createBulkString(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "Content must not be null");
+        if (!content.isReadable()) {
+            return EMPTY_BULK_STRING;
+        }
+        return new RefCountedBulkStringRedisMessage(content);
+    }
+
+    /**
+     * Creates a {@link ArrayHeaderRedisMessage} for the given {@code length}.
+     *
+     * @return a {@link ArrayHeaderRedisMessage} for the given {@code length}.
+     */
+    public static RedisMessage createArrayHeader(long length) {
+        return new ArrayHeaderRedisMessage(length);
+    }
+
+    /**
+     * Creates a {@link ArrayRedisMessage} with {@code null} content.
+     *
+     * @return a {@link ArrayRedisMessage} with {@code null} content.
+     */
+    public static RedisMessage nullArray() {
+        return NULL_ARRAY;
+    }
+
+    /**
+     * Creates a {@link ArrayRedisMessage} with empty content.
+     *
+     * @return a {@link ArrayRedisMessage} with empty content.
+     */
+    public static RedisMessage emptyArray() {
+        return EMPTY_ARRAY;
+    }
+
+    /**
+     * Creates a {@link ArrayRedisMessage} for the given {@code content}.
+     * Note that non-empty {@code content} results in {@link io.netty.util.ReferenceCounted} instances.
+     *
+     * @param content the content, must not be {@code null}.
+     * @return a {@link ArrayRedisMessage} for the given {@code content}.
+     */
+    public static RedisMessage createArray(List<RedisMessage> content) {
+        ObjectUtil.checkNotNull(content, "Content must not be null");
+        if (content.isEmpty()) {
+            return EMPTY_ARRAY;
+        } else {
+            return new RefCountedArrayRedisMessage(content);
+        }
+    }
+
+    private static RedisMessage findPredefinedString(PreloadString[] predefinedStrings, String content) {
+        for (PreloadString predefined : predefinedStrings) {
+            if (predefined.content.equals(content)) {
+                return predefined.message;
+            }
+        }
+        return null;
+    }
+
+    private static final class PreloadString {
+        private final String content;
+        private final ByteBuf byteBuf;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "byteBuf" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1068)